python/graphstorm/dataloading/dataset.py

```diff
@@ -436,31 +441,52 @@ def prepare_data(self, g):
         """
         pb = g.get_partition_book()
         test_idxs = {}
+        infer_idxs = {}
         # If eval_etypes is None, we use all edge types.
         if self.eval_etypes is None:
             self._eval_etypes = g.canonical_etypes
-        # test_mask exists
         for canonical_etype in self.eval_etypes:
             if 'test_mask' in g.edges[canonical_etype].data:
+                # test_mask exists
+                # we will do evaluation.
                 test_idx = dgl.distributed.edge_split(
                     g.edges[canonical_etype].data['test_mask'],
                     pb, etype=canonical_etype, force_even=True)
                 # If there are test data globally, we should add them to the dict.
                 if test_idx is not None and dist_sum(len(test_idx)) > 0:
                     test_idxs[canonical_etype] = test_idx
-                elif test_idx is None:
-                    print(f"WARNING: {canonical_etype} does not contains " \
-                            "test data, skip testing {canonical_etype}")
+            elif 'infer_mask' in g.edges[canonical_etype].data:
```

Comment:
we are going to support a new type of mask?

Reply:
I think this can provide more flexibility for users to specify a subset of edges to do inference.